### ISSUE TITLE
Why preload.ts in main folder instead of render?

### ISSUE THREAD

**Issue by davidliudev** [electron-react-boilerplate/electron-react-boilerplate#3561]
As my understanding, the preload.ts is executed in the renderer process during every BrowserWindow initialisation rather than in the main process.
Does putting it in the main folder wrongly suggest that it is running in the main process?

Shouldn't we move it to the render folder instead to make it less misleading?

Correct me if I am wrong.

**Comment by aolsenjazz** [comment]
According to the [electron docs](https://www.electronjs.org/docs/latest/tutorial/tutorial-preload), the preload file runs in a sandboxed process with access to the DOM and a limited subset of node API's. I agree that having it in the main folder seems to suggest that it's a part of the main process, but given that it exists in neither the main nor renderer processes, it seems there is no "perfect" home for this file.

As to your suggestion, it seems to me that moving it to the renderer would have the exact same effect w.r.t. where it belongs.

**Comment by davidliudev** [comment]
Actually from the doc you linked it suggests that this preload script runs under the renderer process since it has both access to the node env(of the renderer process) and the DOM(of the renderer process). It does runs under a separate context though.

The IpcRender that we usually inject from the preload script comes from the node part of the renderer process. Actually if we don’t enable security isolation you can access it directly without the needs of a preload script.


I used to understand it wrongly before and later introduced many bugs to my project, especially when dealing with node native modules. 

Correct me if I’m wrong.

**Comment by davidliudev** [comment]
Moreover since we can attach different preload script to different BrowserWindow it suggest that the script shall follow the GUI part respectively, rather than in Main.

**Comment by aolsenjazz** [comment]
This concept gave me a lot of trouble too. You're definitely correct in that it's injected into the renderer. However, my takeaway from these docs is that the electron team doesn't want devs thinking of this as a component of the renderer. It seems that they've taken care to express this as a different concept from renderer/main, with its own set of duties and available subsets of APIs.

Certainly nothing to correct here - I think everything you've said checks out. I think the difference is the package maintainers' intention. Also still agree that living in main doesn't feel _correct_.

**Comment by HermanL02** [comment]
I totally agree with your opinion here. I was literally confused when I follow the official ipc instruction.
